packages/firebase_remote_config/firebase_remote_config_platform_interface/lib/src/method_channel/utils/exception.dart

```diff
@@ -0,0 +1,35 @@
+import 'package:firebase_core/firebase_core.dart';
+import 'package:flutter/services.dart';
+
+/// Catches a [PlatformException] and converts it into a [FirebaseException] if
+/// it was intentionally caught on the native platform.
+Exception convertPlatformException(Object exception) {
+  if (exception is! Exception || exception is! PlatformException) {
+    throw exception;
+  }
+
+  return platformExceptionToFirebaseException(exception as PlatformException);
+}
+
+/// Converts a [PlatformException] into a [FirebaseException].
+///
+/// A [PlatformException] can only be converted to a [FirebaseException] if the
+/// `details` of the exception exist. Firebase returns specific codes and messages
+/// which can be converted into user friendly exceptions.
+FirebaseException platformExceptionToFirebaseException(
+    PlatformException platformException) {
+  Map<String, String> details = platformException.details != null
+      ? Map<String, dynamic>.from(platformException.details)
+      : null;
+
+  String code = 'unknown';
+  String message = platformException.message;
+
+  if (details != null) {
+    code = details['code'] ?? code;
+    message = details['message'] ?? message;
+  }
+
+  return FirebaseException(
```

Comment:
See previous comment, `FirebaseException` now accepts a `stackTrace`.

Reply:
Used `StackTrace.current`, is that the intension/expectation?

Reply:
I'm not sure if the intension is to get the native stack trace and have that available on the Dart side. Let me know.

Reply:
It's best coming from the actual exception, example in above comment on how to do that

Reply:
Done